src/components/inputs/async-select-input/async-select-input.js

```diff
@@ -139,5 +147,9 @@ export class AsyncSelectInput extends React.Component {
 }
 
 const Wrapped = injectIntl(AsyncSelectInput);
-Wrapped.isTouched = AsyncSelectInput.isTouched;
+addStaticFields(Wrapped, {
+  ...defaultComponents,
+  ...customizedComponents,
```

Comment:
Wouldn't these effectively overwrite each other in case they use the same name?

To me it would be a bit clearer if we used names again for the static exports, so:

```js
addStaticFields(Wrapped, {
  defaultComponents,
  components,
  isTouched: /*  ..*/
})
```

I also likes that this groups the components, so that it's clear which ones we provided, and which ones are official from `react-select`. And I like that this does not pollute the statics too much.

What do you think?

Reply:
@dferber90 I am personally think `<SelectInput.Option>` looks way better in code then `<SelectInput.defaultComponents.Option>`. But that's good that you actually raised that question, because, now I think we should not export components from `components` at all, because anyways we are not letting to customise them by doing that:

```js
<AsyncSelect
...
  components={{
    ...this.props.components,
    ...customizedComponents,
  }}
...
/>
```

So why not ending up with something like:

```
addStaticFields(Wrapped, {
  ...omit(defaultComponents, Object.keys(components),
  isTouched: /*  ..*/
})
```

Reply:
I would have done this the other way around as

Assuming that `customizedComponents` are the components we overwrite by default, I would have assumed we do:

```jsx
<AsyncSelect
// ...
  components={{
    ...customizedComponents,
    ...this.props.components,
  }}
// ...
/>
```

Then users can even overwrite our customization with the default again in case they need to.


Regarding the exports, I think users could do this instead

```js
const { Option } = SelectInput.defaultComponents
const CustomOption = () => <Option />
```

I'm open to all approaches! We can always adjust it as we go. Feel free to either merge or use the way I suggested above.